reverse_engineering/helpers/clusterHelper.js

```diff
@@ -401,21 +401,21 @@ const getDbCollectionData = async ({
 	});
 	const documentKind = data.documentKinds?.[bucketName]?.documentKindName || '';
 	const options = { limit, pagination: data.pagination, bucketName, scopeName, collectionName };
+	const fieldInference = data.fieldInference;
 
 	let documents = [];
+	let standardDocument = null;
 	let query = queryHelper.getSelectCollectionDocumentsQuery({ bucketName, scopeName, collectionName });
 
 	try {
 		documents = await getPaginatedQuery({ cluster, options, query, logger });
-
-		return schemaHelper.getDbCollectionData({
+		standardDocument = await getCollectionDocumentByDocumentId({
+			cluster,
 			bucketName,
 			scopeName,
 			collectionName,
-			documentKind,
-			documents,
-			collectionIndexes,
-			includeEmptyCollection,
+			documentId: documents[0]?.docid,
+			logger,
 		});
 	} catch (error) {
 		try {
```

Comment:
It's hard to read the catch block. Could you extract the code from the catch block into a separate function.